aptos-move/aptos-e2e-comparison-testing/src/lib.rs

```diff
@@ -528,23 +552,47 @@ fn dump_and_compile_from_package_metadata(
     std::fs::write(toml_path, manifest.to_string()).unwrap();
 
     // step 5: test whether the code can be compiled
-    if let std::collections::hash_map::Entry::Vacant(e) = compilation_cache
+    if !compilation_cache
         .compiled_package_map
-        .entry(package_info.clone())
+        .contains_key(&package_info)
     {
-        let mut build_options = BuildOptions::default();
-        build_options
-            .named_addresses
-            .insert(package_info.package_name.clone(), package_info.address);
-        build_options.compiler_version = compiler_verion;
-        let compiled_package = BuiltPackage::build(root_package_dir, build_options);
-        if let Ok(built_package) = compiled_package {
-            e.insert(built_package.package);
+        let package_v1 = compile_package(
+            root_package_dir.clone(),
+            &package_info,
+            Some(CompilerVersion::V1),
+        );
+        if let Ok(built_package) = package_v1 {
+            if execution_mode.is_some() && execution_mode.unwrap().is_v1_or_compare() {
```

Comment:
nit: you can use `is_some_and` to merge this &&

Reply:
done